src/rdo.rs

```diff
@@ -1124,7 +1124,16 @@ fn inter_frame_rdo_mode_decision<T: Pixel>(
           true,
         ));
         for &x in RAV1E_INTER_COMPOUND_MODES {
-          inter_mode_set.push((x, ref_frames_set.len() - 1));
+          // exclude any NEAR mode based on speed setting
+          if fi.config.speed_settings.include_near_mvs || !x.has_nearmv() {
+            let mv_stack_idx = ref_frames_set.len() - 1;
+            // exclude NEAR modes if the mv_stack is too short
+            if !(x.has_nearmv()
+              && x.ref_mv_idx() >= mv_stacks[mv_stack_idx].len())
```

Comment:
The push of `mv_stack` into `mv_stacks` occurs after this loop. It is not clear that `mv_stack_idx`is the right index here anyhow.
```suggestion
              && x.ref_mv_idx() >= mv_stack.len())
```

Reply:
Indeed. I think I broke this when changing it to exclude NEAR.